devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy

```diff
@@ -0,0 +1,337 @@
+	return result
+}
+
+def installed() {
+	refresh()
+}
+
+def refresh() {
+	return zigbee.readAttribute(THERMOSTAT_CLUSTER, LOCAL_TEMPERATURE) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, COOLING_SETPOINT) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, THERMOSTAT_MODE) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, THERMOSTAT_RUNNING_STATE) +
+			zigbee.readAttribute(FAN_CONTROL_CLUSTER, FAN_MODE)
+}
+
+def ping() {
+	refresh()
+}
+
+def configure() {
+	def binding = zigbee.addBinding(THERMOSTAT_CLUSTER) + zigbee.addBinding(FAN_CONTROL_CLUSTER)
+	def startValues = zigbee.writeAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT, DataType.INT16, 0x0A28) +
+			zigbee.writeAttribute(THERMOSTAT_CLUSTER, COOLING_SETPOINT, DataType.INT16, 0x07D0)
+
+	return binding + startValues + refresh()
+}
+
+
+def getTemperature(value) {
+	if (value != null) {
+		def celsius = Integer.parseInt(value, 16) / 100
+		if (temperatureScale == "C") {
+			return celsius
+		} else {
+			return Math.round(celsiusToFahrenheit(celsius))
+		}
+	}
+}
+
+def setThermostatMode(value) {
+	switch(value) {
+		case "heat":
+			heat()
+			break
+		case "cool":
+			cool()
+			break
+		default:
+			off()
+	}
+}
+
+def off() {
+	return zigbee.writeAttribute(THERMOSTAT_CLUSTER, THERMOSTAT_MODE, DataType.ENUM8, THERMOSTAT_MODE_OFF) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, THERMOSTAT_MODE)
+}
+
+def cool() {
+	return zigbee.writeAttribute(THERMOSTAT_CLUSTER, THERMOSTAT_MODE, DataType.ENUM8, THERMOSTAT_MODE_COOL) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, THERMOSTAT_MODE)
+}
+
+def heat() {
+	return zigbee.writeAttribute(THERMOSTAT_CLUSTER, THERMOSTAT_MODE, DataType.ENUM8, THERMOSTAT_MODE_HEAT) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, THERMOSTAT_MODE)
+}
+
+def fanAuto() {
+	return zigbee.writeAttribute(FAN_CONTROL_CLUSTER, FAN_MODE, DataType.ENUM8, FAN_MODE_AUTO) +
+			zigbee.readAttribute(FAN_CONTROL_CLUSTER, FAN_MODE)
+}
+
+def fanOn() {
+	return zigbee.writeAttribute(FAN_CONTROL_CLUSTER, FAN_MODE, DataType.ENUM8, FAN_MODE_ON) +
+			zigbee.readAttribute(FAN_CONTROL_CLUSTER, FAN_MODE)
+}
+
+def setCoolingSetpoint(degrees) {
+	if (degrees != null) {
+		def degreesInteger = Math.round(degrees)
+		def celsius = (temperatureScale == "C") ? degreesInteger : (fahrenheitToCelsius(degreesInteger) as Double).round(2)
+		return zigbee.writeAttribute(THERMOSTAT_CLUSTER, COOLING_SETPOINT, DataType.INT16, hex(celsius * 100)) +
+				zigbee.readAttribute(THERMOSTAT_CLUSTER, COOLING_SETPOINT)
+	}
+}
+
+def setHeatingSetpoint(degrees) {
+	if (degrees != null) {
+		def degreesInteger = Math.round(degrees)
+
+		def celsius = (temperatureScale == "C") ? degreesInteger : (fahrenheitToCelsius(degreesInteger) as Double).round(2)
+		return zigbee.writeAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT, DataType.INT16, hex(celsius * 100)) +
+				zigbee.readAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT)
+	}
+}
+
+def raiseHeatingSetpoint() {
```

Comment:
Some of our other thermostat DTHs have a delay here before sending the actual command in order to collect multiple arrow presses into just one set instead of several "up by 1"s.

I know OneApp doesn't work like this, but I think it's worth doing for the classic DTH for a better user experience.

If it's too complicated and error-prone that you don't think it's worth it, I'd understand that too.